app/Services/GraphQL.php

```diff
@@ -15,14 +16,50 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client
+     * and parse the data result into a convenient format.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @param  $queryKey  String
+     * @return array|null
      */
-    public function query($query, $variables)
+    public function query($query, $variables, $queryKey)
     {
-        return $this->client->query($query, $variables);
+        // Use try/catch to avoid any GraphQL related errors breaking the application.
+        try {
+            $response = $this->client->query($query, $variables);
+        } catch (\Exception $exception) {
+            Log::error(
+                'GraphQL request failed. Variables: '.json_encode($variables).' Exception: '.$exception->getMessage()
```

Comment:
How's this log? Should we perhaps include the `$query` itself as well?

Reply:
Yeah, I think including the query (or if it's parsed anywhere convenient, the query name) makes sense! 👍

Reply:
Added the `$query` to the log output in https://github.com/DoSomething/rogue/pull/889/commits/1d5fc99d6ea1cd09e68c41ff59b8133e087dcfa0